### PR TITLE
[MNOE-1110][OPAL-431]Add subscription event attributes

### PR DESCRIPTION
**Pull request by iseessel**

### CODE REVIEW

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb [hunk 1]
```diff
@@ -39,16 +39,25 @@ def create
       .first
     return render_not_found('Organization') unless organization
 
-    subscription = MnoEnterprise::Subscription.new(subscription_update_params)
+    if cart_subscription_param.present?
+      # Workaround, because once a subscription is requested from the cart, the attributes of the subscription get
+      # deleted and moved over to the subscription event. TODO: Refactor cart system to take place on subscription event.
+      subscription = MnoEnterprise::Subscription.new(subscription_cart_params)
+      subscription.status = :staged
```
Comment (adamaziz15) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb]: This is repeated on line 51

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb [hunk 2]
```diff
@@ -39,16 +39,25 @@ def create
...
+    else
+      subscription = MnoEnterprise::Subscription.new(subscription_update_params)
+    end
+
     subscription.status = :staged if cart_subscription_param.present?
     subscription.relationships.organization = organization
-    if params[:subscription][:currency]
-      subscription.currency = params[:subscription][:currency]
```
Comment (adamaziz15) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb]: Do we not need to set the currency here anymore?

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/subscriptions_controller.rb [hunk 3]
```diff
@@ -120,14 +125,25 @@ def cart_subscription_param
...
     # custom_data is an arbitrary hash
     # On Rails 5.1 use `permit(custom_data: {})`
-    params.require(:subscription).permit(:start_date, :max_licenses, :product_pricing_id, :product_contract_id, :custom_data).tap do |whitelisted|
+    subscription_params.permit(:start_date, :max_licenses, :product_pricing_id, :product_contract_id, :custom_data, :currency).tap do |whitelisted|
       whitelisted[:custom_data] = params[:subscription][:custom_data] if params[:subscription].has_key?(:custom_data) && params[:subscription][:custom_data].is_a?(Hash)
     end
   end
 
+  def subscription_update_params
+    # custom_data is an arbitrary hash
```
Comment (adamaziz15) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/subscriptions_controller.rb]: Dont need this comment anymore

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb [hunk 4]
```diff
@@ -88,14 +98,25 @@ def cart_subscription_param
...
     # custom_data is an arbitrary hash
     # On Rails 5.1 use `permit(custom_data: {})`
-    params.require(:subscription).permit(:start_date, :max_licenses, :custom_data, :product_contract_id, :product_pricing_id).tap do |whitelisted|
+    subscription_params.permit(:start_date, :max_licenses,:custom_data, :product_contract_id, :product_pricing_id, :currency).tap do |whitelisted|
       whitelisted[:custom_data] = params[:subscription][:custom_data] if params[:subscription].has_key?(:custom_data) && params[:subscription][:custom_data].is_a?(Hash)
     end
   end
 
+  def subscription_update_params
+    # custom_data is an arbitrary hash
```
Comment (adamaziz15) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb]: Dont think we need this comment anymore

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb (hunk 1 above)
Comment (iseessel) [reply api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/subscriptions_controller.rb]: got it

**Comment by adamaziz15** [comment]
@iseessel Forgot the locales from mnoe-ng

**Comment by iseessel** [comment]
@adamaziz15 Fixed.

File: api/app/views/mno_enterprise/jpi/v1/admin/product_pricing/_product_pricing.json.jbuilder [hunk 5]
```diff
@@ -0,0 +1 @@
+json.extract! product_pricing, :id, :name, :description, :free, :license_based, :pricing_type, :free_trial_enabled, :free_trial_duration, :free_trial_unit, :position, :per_duration, :per_unit, :prices, :external_id
```
Comment (paulstuartparker) [inline api/app/views/mno_enterprise/jpi/v1/admin/product_pricing/_product_pricing.json.jbuilder]: :quote_based

File: api/app/views/mno_enterprise/jpi/v1/product_pricing/_product_pricing.json.jbuilder [hunk 6]
```diff
@@ -0,0 +1 @@
+json.extract! product_pricing, :id, :name, :description, :free, :license_based, :pricing_type, :free_trial_enabled, :free_trial_duration, :free_trial_unit, :position, :per_duration, :per_unit, :prices, :external_id
```
Comment (paulstuartparker) [inline api/app/views/mno_enterprise/jpi/v1/product_pricing/_product_pricing.json.jbuilder]: :quote_based

**Pull request merged**